lib/Dialect/FIRRTL/Transforms/LowerDomains.cpp

```diff
@@ -307,11 +310,22 @@ LogicalResult LowerCircuit::lowerDomain(DomainOp op) {
   ImplicitLocOpBuilder builder(op.getLoc(), op);
   auto *context = op.getContext();
   auto name = op.getNameAttr();
-  // TODO: Update this once DomainOps have properties.
-  auto classIn = ClassOp::create(builder, name, {});
+
+  // Create the new input class.  This is what the user will need to specify to
```

Comment:
Could the class just take the body as-is, instead of cloning?

.... do we even need to replace the domain with a class?

Reply:
Ah, good point. This can just be `takeBody`. Switched in: 283d2194f

Reply:
Wrong commit. Fixed in: https://github.com/llvm/circt/pull/9032/commits/1d6774cd8cf928020bd1e59d078a1422c1c02fc4